File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int max(int x,int y){
        return (x>y)?x:y;
    }
    
    int maxProfit(vector<int>& prices) {
        int l=0;
        int r=0;
        int global_max=0;
        
        while(r<prices.size()){
            if(prices[l]<prices[r]){
                global_max=max(global_max,prices[r]-prices[l]);
                r++;
            }else{
                l=r;
                r++;
            }
        }
        
        return global_max;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l=0
        h=0
        n=len(prices)
        mx=0
        while(h<n):
            mx=max(mx,prices[h]-prices[l])
            if (prices[h]>=prices[l]):
                h+=1
            else:
                l=h
                h+=1
        return mx
            